sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteBundlesToFiles.java

```diff
@@ -79,6 +80,7 @@
     public final DestinationT destination;
 
     public Result(String filename, Long fileByteSize, DestinationT destination) {
+      checkNotNull(destination);
```

Comment:
Next time, I'd suggest `this.destination = checkNotNull(destination)`.

Sholudn't the filename and fileByteSize also be non-nullable?